packages/sdk-js/src/DidHelpers/checkResult.ts

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright (c) 2018-2024, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+/* eslint-disable jsdoc/require-jsdoc */
+// functions in this file are not meant to be public
+
+import type { ApiPromise } from '@polkadot/api'
+import type { SubmittableResultValue } from '@polkadot/api/types'
+import type { BlockNumber } from '@polkadot/types/interfaces'
+import { u8aToHex, u8aToU8a } from '@polkadot/util'
+import type { FrameSystemEventRecord as EventRecord } from '@kiltprotocol/augment-api'
+import { resolver as DidResolver, signersForDid } from '@kiltprotocol/did'
+import type { Did, HexString } from '@kiltprotocol/types'
+import type { SharedArguments, TransactionResult } from './interfaces.js'
+import { assertStatus, mapError } from './common.js'
+
+function checkEventsForErrors(
+  api: ApiPromise,
+  txEvents: EventRecord[] = []
+): Error | undefined {
+  let error
+  txEvents.forEach(({ event }) => {
```

Comment:
```suggestion
  return txEvents.find(({ event }) => {
```
combine this with returns (I'll make one example below) and our function becomes simpler and faster (exits on first error)